<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <remap from="/usb_cam/image_raw" to="/image"/>
  </node>
    <node pkg="turtlesim" name="turtle1" type="turtle_teleop_key" required="true" launch-prefix="xterm -e">
        <remap from="turtle1/cmd_vel" to="turtle1/cmd_vel_pid" />
    </node>
    <node name="balboa_pid" pkg="balboa_core" type="balboa_pid.py" launch-prefix="xterm -e"/>
    <node name="balboa_angularvelpid" pkg="balboa_core" type="balboa_angularvelpid.py"/>
    <node name="balboa_teleop_pid" pkg="balboa_core" type="balboa_teleop_pid.py"/>
    <node name="balboa_teleop" pkg="balboa_core" type="balboa_teleop.py"/>
    <node name="balboa_serial" pkg="balboa_core" type="balboa_serial.py"/>

    <rosparam param="thresh/">
    high: {h: 15, s: 255, v: 255}
    low: {h: 0, s: 100, v: 100}
    </rosparam>
    <node name="ballDetector" pkg="ball_detector" type="ball_detector_node"/>
</launch>
